File: java-api-core/src/com/kinvey/java/sync/RequestMethod.java
package com.kinvey.java.sync;

/**
 * Created by yuliya on 09/28/17.
 */

public enum RequestMethod {
    SAVE("SAVE"),
    DELETE("DELETE");

    private String query;

    RequestMethod(String query) {
        this.query = query;
    }

    public static RequestMethod fromString(String verb){
        for (RequestMethod v : RequestMethod.values()){
            if (v.query.equals(verb)){
                return v;
            }
        }
        return null;
    }
}

File: java-api-core/src/com/kinvey/java/sync/RequestMethod.kt
package com.kinvey.java.sync

/**
 * Created by yuliya on 09/28/17.
 */

enum class RequestMethod (private val query: String) {
    SAVE("SAVE"),
    DELETE("DELETE");

    companion object {

        @JvmStatic
        fun fromString(verb: String): RequestMethod? {
            for (v in values()) {
                if (v.query == verb) {
                    return v
                }
            }
            return null
        }
    }
}
